### PR TITLE
Add Usage command and features

### PR DESCRIPTION
**Pull request by ChesterSim**

### CODE REVIEW

File: src/main/java/csdev/couponstash/logic/commands/UsedCommand.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    private static Coupon createUsedCoupon(Coupon couponToBeUsed) {
+        Name name = couponToBeUsed.getName();
+        Phone phone = couponToBeUsed.getPhone();
+        Set<Tag> tags = couponToBeUsed.getTags();
+        Usage updatedUsage = couponToBeUsed.getUsage().increaseUsageByOne();
+
+        return new Coupon(name, phone, updatedUsage, tags);
+    }
+}
```
Comment (alcen) [inline src/main/java/csdev/couponstash/logic/commands/UsedCommand.java]: Should have a new line at the end of the file (idk why)

File: src/main/java/csdev/couponstash/model/coupon/Usage.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                || (other instanceof Usage
+                && value.equals(((Usage) other).value))
+                && maxUsage.equals(((Usage) other).maxUsage);
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+}
```
Comment (alcen) [inline src/main/java/csdev/couponstash/model/coupon/Usage.java]: Again, should have new line at end of file

**Review by alcen — COMMENTED** [review]
Nice work!! But Travis complains about some checkstyle issues (see comments)

**Review by Denise-Ng — APPROVED** [review]
Okay LGTM

**Comment by nelsontky** [comment]
> Okay LGTM

Is LGTM some new age slang? Sorry I'm a boomer

**Comment by ChesterSim** [comment]
@nelsontky it means Looks Good To Me!

**Pull request merged**